rust/lance/src/datafusion/physical_expr.rs

```diff
@@ -129,6 +129,13 @@ pub fn column_names_in_expr(expr: &dyn PhysicalExpr) -> Vec<String> {
     visitor.columns
 }
 
+pub fn unique_column_names_in_expr(expr: &dyn PhysicalExpr) -> Vec<String> {
+    let mut cols = column_names_in_expr(expr);
+    cols.sort();
+    cols.dedup();
+    cols
+}
```

Comment:
Alternatively, we could just make this the default behavior of `column_names_in_expr`.  thoughts @wjones127 ?

Reply:
In https://github.com/lancedb/lance/pull/1581, I change `column_names_in_expr` to return distinct columns already. I think that's what we always want. Let me know what you think.

Reply:
Yep, awesome.  I'll take a look later.